tests/foreman/ui/test_ansible.py

```diff
@@ -449,39 +446,48 @@ def test_positive_assign_and_remove_ansible_role_to_hostgroup(
 
     @pytest.mark.tier3
     def test_positive_non_admin_user_access_with_usergroup(
-        self, request, target_sat, test_name, module_org, module_location
+        self,
+        request,
+        target_sat,
+        test_name,
     ):
         """Verify non-admin user can access the ansible page on WebUI
 
         :id: 82d30664-1b74-457c-92e2-31a5ba89e826
 
-        :BZ: 2158508
-
         :steps:
             1. Create user with non-admin
             2. Create usergroup with administrator role
             3. Log in as a user and try to access WebUI -> Hosts -> select host -> Ansible
+            4. Assign ansible role to the host
+
+        :expectedresults: The user is able to view the Ansible page and assign roles because they are an administrator due to user group
+
+        :BZ: 2158508
 
-        :expectedresults: See Ansible page as user should be administrator due to usergroup
+        :Verifies: SAT-15826
 
         :customerscenario: true
         """
         SELECTED_ROLE = 'RedHatInsights.insights-client'
+        name = gen_string('alpha')
         password = gen_string('alpha')
-        user = target_sat.api.User(
-            login=gen_string('alpha'), password=password, admin=False
+        admin_user = target_sat.api.User(
```

Comment:
You can remove the role from user and logout the session. I am guessing that an open user session is causing the conflict in the teardown. After that, you should be able to delete the user.

Reply:
https://github.com/SatelliteQE/robottelo/commit/919bb87a74f566b052cca7dae17be39e2723d2a5 I tried your suggestion, but that wasn't the issue. When I log in as a non-admin user and the owner is not set to Satellite, assigning a role to the sat causes the non-admin user to become the owner, even after I remove the role. I am still facing the same teardown issue, even after logging out. To prevent this, I created the host and assigned the admin as the owner, so when the non-admin user logs in and performs certain actions, it won’t affect the teardown process later.